### DISCUSSION TITLE
Theme changing after metals starts

### DISCUSSION THREAD

**Discussion by ex0ns** [scalameta/nvim-metals#603] [Q&A] [answered]
Hi ! 
it might not be related to metals, but I think it could be, when I open neovim the color scheme is as I expect, however once metals is loaded the color changes and are way less useful that what they are at boot, I don't know how to change that and I could not figure out where this is configured.

is there something that I could change inside metals' options in order to not change the colors of the token when loaded ?

https://github.com/scalameta/nvim-metals/assets/1064129/beea4d79-f2c5-4fb3-859c-c0d52de03673

If this is not something directly related to nvim-metals, do any of you have an idea of where this is coming from ?

Thanks

**Comment by ckipp01** [comment, marked as answer]
Hey @ex0ns! This is a metals feature called semantic tokens. It used to be off by default, but it has recently been turned on by default in Metals. If you want to turn it off you can set `enableSemanticHighlighting = false` in your config settings. I do that as well since I prefer the treesitter highlighting. With that being said, what you're seeing actually looks like a bug in Metals. I'd recommend actually opening an issue upstream in Metals with a minimal example showing this since that's obviously not what you want to see 😆 .

**Reply by pepper-lees** [reply]
Hey, I'm experiencing the same issue. Here's my bit that loads Metals inside `~/.config/nvim/lua/user/plugins/user.lua`:

```lua
  {
    "scalameta/nvim-metals",
    dependencies = "nvim-lua/plenary.nvim",
    init = function()
      vim.api.nvim_create_autocmd("FileType", {
        pattern = { "scala", "sbt", "java" },
        callback = function()
          local metals = require "metals"
          local config = require("astronvim.utils.lsp").config "metals"
          config.find_root_dir_max_project_nesting = 2
          config.enableSemanticHighlighting = false
          metals.initialize_or_attach(config)
        end,
        group = vim.api.nvim_create_augroup("nvim-metals", { clear = true }),
      })
    end,
  },
```
I added `config.enableSemanticHighlighting = false` after noticing the same thing as OP, where the syntax highlighting becomes much worse after Metals loads, but it didn't change the behavior.

Note: I'm pretty sure that the config is being attached correctly, because the line `config.find_root_dir_max_project_nesting = 2` DOES have an impact on nvim-metals' behavior. (if I set it to 1, then I get a "want to import the new sbt workspace?" dialog)

EDIT: my bad! I misunderstood the documentation. It's working now after I changed the relevant line to this:
```lua
          config.settings = {
            enableSemanticHighlighting = false,
          }
```

**Reply by ex0ns** [reply]
I could have posted my config for future reference, sorry for that, I'm glad that you found the solution:
```lua

  {
    "scalameta/nvim-metals",
    ft = { "scala", "sbt", "java", "sc"},
    dependencies = {
      "nvim-lua/plenary.nvim",
      "mfussenegger/nvim-dap",
    },
    config = function()
      require "custom.configs.metals"
    end,
  },
```

and in `custom/configs/metals.lua`

```lua
local present, metals = pcall(require, "metals")

if not present then
  return
end

local metals_config = metals.bare_config()
local augroup = vim.api.nvim_create_augroup("LspFormatting", {})

metals_config.init_options.statusBarProvider = "on"
metals_config.settings.enableSemanticHighlighting = false

metals_config.on_attach = function(_, bufnr)
  metals.setup_dap()
  vim.api.nvim_create_autocmd("bufwritepost", {
    group = augroup,
    buffer = bufnr,
    callback = function()
      vim.lsp.buf.format()
    end,
  })
end

metals.initialize_or_attach(metals_config)
```

It might not be the best way of doing that, but it works for me

**Comment by ex0ns** [comment]
This was exactly this settings, thanks ! 
I saw it in the documentation but as it seems to default to `false` I did not look much further, thanks for the fix

And regarding the example, I agree that using protoquill codebase as an example might not be the best reproducer, but this was what I was working on as of writing, I will see if I can reproduce on other smaller project and maybe open an issue in metals !

**Reply by ckipp01** [reply]
Thats a good reminder to update the docs since this is no longer the case. I'm on holiday at the moment so I'll do this when I return or please do feel free to shoot in a PR!
